Packages: Move packages repository into Gutenberg

omarreiss opened the pull request:
## Description
<!-- Please describe what you have changed or added -->

Adds https://github.com/WordPress/packages/ including all commit history to Gutenberg.

Once this is merged, we need to start moving over issues and PR's from the old repo to Gutenberg as well.

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

I have only tested the script for creating symlinks. Hoping @gziolo could take over from here. 😄 

## Types of changes
<!-- What types of changes does your code introduce?  -->

I ran `git subtree add --prefix temp-packages https://github.com/WordPress/packages.git master` to preserve the commit history. Then I moved the files over to the right directories, synchronized the config files and scripts and removed what was no longer needed.

<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

adamsilverstein committed:
camelCase for readme variable names

adamsilverstein committed:
update hook and `vendor/plugin/function` wording, removing `Name`

adamsilverstein committed:
complte namespace shortening to `vendor/plugin/function`

adamsilverstein committed:
Merge pull request #13 from adamsilverstein/add-wp-hooks

Add the WordPress hook library from 21170-core in preparation for beta npm release

adamsilverstein committed:
Merge branch 'master' of github.com:WordPress/packages

adamsilverstein committed:
Publish

 - @wordpress/a11y@0.1.0-beta.2
 - @wordpress/dom-ready@0.1.0-beta.2
 - @wordpress/hooks@0.1.0-beta.0
 - @wordpress/url@0.1.0-beta.2

adamsilverstein committed:
add publishConfig->access:public for hooks

adamsilverstein committed:
Merge branch 'make-hooks-public'

adamsilverstein committed:
Publish

 - @wordpress/hooks@0.1.0-beta.1

adamsilverstein committed:
correctly enable dashes in regexes in validateNamespace

adamsilverstein committed:
Export a hooks function enabling adding hooks via composition

adamsilverstein committed:
test hooks can be instantiated

adamsilverstein committed:
Run test on objects instead of globally

adamsilverstein committed:
correct data return for Hooks

adamsilverstein committed:
Rework the Hooks object, use a local HOOKS instead of global

adamsilverstein committed:
use object based hooks throughout testing

adamsilverstein committed:
remove unused global HOOKS

adamsilverstein committed:
remove some logging

adamsilverstein committed:
remove extra parts from namespace in tests

adamsilverstein committed:
remov vendor/plugin prefix requirement for namespaces

adamsilverstein committed:
update docs

adamsilverstein committed:
Add a test for “Can add filters with dashes in namespaces”

adamsilverstein committed:
Merge pull request #39 from WordPress/fix/correct-namespace-check-for-hooks

Correctly enable dashes in regexes in validateNamespace

aduth committed:
a11y: Apply .screen-reader-text styles to container

aduth committed:
a11y: Update tests per updated style

Questionable value of the test assertions, which are specific to the implementation and not the expected intent of the test case

adamsilverstein committed:
update readme, add some tests

adamsilverstein committed:
Merge branch 'master' of github.com:WordPress/packages

adamsilverstein committed:
Merge branch 'master' into add-hooks-via-composition

# Conflicts:
#	packages/hooks/src/test/index.test.js
#	packages/hooks/src/validateNamespace.js

adamsilverstein committed:
switch naming to `createHooks` and correct tests

adamsilverstein committed:
remove extraneous single quotes

adamsilverstein committed:
remove the HOOKS global abstraction

adamsilverstein committed:
update test beforeEach cleanup to match new object structure

adamsilverstein committed:
use export defaults

adamsilverstein committed:
Clean up readme for new approach

adamsilverstein committed:
url - add built files

adamsilverstein committed:
hooks - add built files

adamsilverstein committed:
dom-ready - add built files

adamsilverstein committed:
a11y - add built files

adamsilverstein committed:
remove build filders from gitignore

adamsilverstein committed:
Merge branch 'add-built-files-for-release'

adamsilverstein committed:
Publish

 - @wordpress/a11y@0.1.0-beta.3
 - @wordpress/dom-ready@0.1.0-beta.3
 - @wordpress/hooks@0.1.0-beta.2
 - @wordpress/url@0.1.0-beta.3

adamsilverstein committed:
re-ignore build files; didn’t resolve publishing issue

adamsilverstein committed:
Simplify dom-ready package.json, add a license field

adamsilverstein committed:
Merge branch 'package-work'

adamsilverstein committed:
Publish

 - @wordpress/a11y@0.1.0-beta.4
 - @wordpress/dom-ready@0.1.0-beta.4
 - @wordpress/hooks@0.1.0-beta.3

adamsilverstein committed:
Update readmes to include `@next` in npm install line

Since our packages are still in pre-release, the @next tag is required to install the latest published version. We can remove this instruction once the packages are past pre-release.

adamsilverstein committed:
Merge branch 'clarify-installation-instructions'

adamsilverstein committed:
Publish

 - @wordpress/a11y@0.1.0-beta.5
 - @wordpress/dom-ready@0.1.0-beta.5
 - @wordpress/hooks@0.1.0-beta.4
 - @wordpress/url@0.1.0-beta.4

adamsilverstein committed:
Merge branch 'master' into add-hooks-via-composition

# Conflicts:
#	packages/hooks/README.md

adamsilverstein committed:
Merge pull request #40 from WordPress/add-hooks-via-composition

Add Hooks to objects via composition or as a mixin

adamsilverstein committed:
Merge branch 'master' of github.com:WordPress/packages

adamsilverstein committed:
remove the built files which were added by mistake

adamsilverstein committed:
Merge branch 'remove-built-files'

adamsilverstein committed:
Publish

 - @wordpress/a11y@0.1.0-beta.6
 - @wordpress/dom-ready@0.1.0-beta.6
 - @wordpress/hooks@0.1.0-beta.5
 - @wordpress/url@0.1.0-beta.5

gziolo committed:
Hooks: Update hooks public API to make it possible to apply to wp.hooks directly (#45)

gziolo committed:
Development: Improve onboarding experience for new contributors (#46)

omarreiss committed:
Merge pull request #42 from WordPress/fix/38-speak-region-overflow

a11y: Use screen-reader-text styles for speak container

afercia committed:
Update screen-reader-text CSS.

omarreiss committed:
Merge pull request #49 from WordPress/update/screen-reader-text-css

Update screen-reader-text CSS

gziolo committed:
Hooks: Allow slashes in the namespace (#47)

* Hooks: Allow slashes in the namespace

* Hooks: Add test ensuring backlash is not allowed inside namespace

adamsilverstein committed:
Merge branch 'master' of github.com:WordPress/packages

adamsilverstein committed:
Publish

 - @wordpress/a11y@1.0.0
 - @wordpress/hooks@1.0.0

adamsilverstein committed:
a11y & hooks: remove ‘@next’ from install instructions

adamsilverstein committed:
Merge pull request #50 from WordPress/fix/update-readme-after-release

a11y & hooks: remove ‘@next’ from install instructions

adamsilverstein committed:
Merge branch 'master' of github.com:WordPress/packages

adamsilverstein committed:
tag a11y and hooks at 1.0.1

adamsilverstein committed:
Merge pull request #51 from WordPress/tag-101-release

tag a11y and hooks at 1.0.1

adamsilverstein committed:
Merge branch 'master' of github.com:WordPress/packages

adamsilverstein committed:
Trigger actions when adding or removing hooks

adamsilverstein committed:
add tests for expected hookAdded and hookRemoved events

adamsilverstein committed:
these are hooks not hoods (typo)

aduth committed:
Implement autop package

aduth committed:
Port PHP wpautop faithfully to JS

gziolo committed:
Tests: Refactor tests to use Jest mocks

gziolo committed:
Standardize format of package.json files (#54)

omarreiss committed:
Merge pull request #3 from ntwb/patch-1

Add GPLv2 `LICENSE.md` license

adamsilverstein committed:
Merge branch 'feature/events-on-add-remove' of github.com:WordPress/packages into feature/events-on-add-remove

adamsilverstein committed:
Add tests for addFilter/removeFilter

adamsilverstein committed:
whitespace

adamsilverstein committed:
Docs update for hooks added/removed events

gziolo committed:
Hooks: Final touches for the add/remove actions

gziolo committed:
Merge pull request #52 from WordPress/feature/events-on-add-remove

Feature/events on add remove

gziolo committed:
Publish

 - @wordpress/a11y@1.0.2
 - @wordpress/dom-ready@0.1.0-beta.7
 - @wordpress/hooks@1.1.0
 - @wordpress/url@0.1.0-beta.6

gziolo committed:
Update Lerna and Jest to the latest version (#56)

aduth committed:
Merge pull request #53 from WordPress/add/wpautop

Add package: autop

ntwb committed:
docs: Update Lerna docs to use `npx`

ntwb committed:
Update license in `package.json` to adhere to SPDX v3.0 specification

ntwb committed:
Update license in `package.json` to adhere to SPDX v3.0 specification

gziolo committed:
Jest-console: Add new package with console object matchers for Jest

gziolo committed:
Jest-console: Add package documentation

gziolo committed:
Jest-console: Integrate test matchers with packages repository

gziolo committed:
Enforce using the latest LTS version (8.x) of Node and up

gziolo committed:
Move node version check to prebuild run script

ntwb committed:
Remove superfluous `a` and use full Node.js name

gziolo committed:
Merge pull request #61 from WordPress/update/node-version

Enforce using the latest LTS version (8.x) of Node and up

gziolo committed:
Jest-console: Fix typos in README file

gziolo committed:
Jest-console: Whitelist folders published to npm

gziolo committed:
Merge pull request #60 from WordPress/add/jest-console

Jest-console: Add new package with console object matchers for Jest

gziolo committed:
Update install instruction in README files that use beta releases (#63)

gziolo committed:
Publish

 - @wordpress/a11y@1.0.3
 - @wordpress/autop@1.0.1
 - @wordpress/dom-ready@1.0.0
 - @wordpress/hooks@1.1.1
 - @wordpress/jest-console@1.0.0
 - @wordpress/url@1.0.0

ntwb committed:
chore: Update `.editorconfig` to match WordPress' upstream

ntwb committed:
chore: Use tabs for indentaion in `lerna.json`

ntwb committed:
chore: Use tabs for indentaion in _root_ `package.json` and `package-lock.json` files

ntwb committed:
chore: Use tabs for indentaion in _url_ package `package.json` and `package-lock.json` files

ntwb committed:
chore: Use tabs for indentaion in _jest-console_ package `package.json` and `package-lock.json` files

ntwb committed:
chore: Use tabs for indentaion in _hooks_ package `package.json` and add missing `package-lock.json` file

ntwb committed:
chore: Use tabs for indentaion in _dom-ready_ package `package.json` and `package-lock.json` files

ntwb committed:
chore: Use tabs for indentaion in _autop_ package `package.json` and `package-lock.json` files

ntwb committed:
chore: Use tabs for indentaion in _a11y_ package `package.json` and `package-lock.json` files

ntwb committed:
chore: Update _autop_ `package.json` fields

gziolo committed:
Babel-preset-default: Add new package containing the default Babel configuration (#66)

gziolo committed:
Publish (#67)

- @wordpress/a11y@1.0.4
 - @wordpress/autop@1.0.2
 - @wordpress/babel-preset-default@1.0.1-0
 - @wordpress/dom-ready@1.0.1
 - @wordpress/hooks@1.1.2
 - @wordpress/jest-console@1.0.1
 - @wordpress/url@1.0.1

ntwb committed:
feat: Add `@wordpress/browserslist-config` package

gziolo committed:
Scripts: New scripts package containing test command (#62)

* Scripts: initial commit with scripts package containing test command

* Publish

 - @wordpress/scripts@1.0.1-0

* Scripts: Minor changes before publishing to npm

gziolo committed:
Publish

 - @wordpress/a11y@1.0.5
 - @wordpress/autop@1.0.3
 - @wordpress/babel-preset-default@1.0.0
 - @wordpress/dom-ready@1.0.2
 - @wordpress/hooks@1.1.3
 - @wordpress/jest-console@1.0.2
 - @wordpress/scripts@1.0.0
 - @wordpress/url@1.0.2

Commit:
📦 NEW: Update Lerna to 2.8.0

ntwb committed:
docs: fix spelling in CHANGELOG.md and consistent formatting

ntwb committed:
chore: remove `browserslist` from `peerDependencies`

ntwb committed:
tests(browserslist-config): refactor test per coding standards

gziolo committed:
Packages: Make sure dependencies are properly exposed for external usage (#71)

gziolo committed:
Publish

 - @wordpress/babel-preset-default@1.0.1
 - @wordpress/browserslist-config@2.0.0
 - @wordpress/jest-console@1.0.3
 - @wordpress/scripts@1.0.1

gziolo committed:
Babel-preset-default: Makre sure transform runtime plugin is not loaded in test env (#73)

gziolo committed:
Publish

 - @wordpress/babel-preset-default@1.0.2

ntwb committed:
chore: use tabs for indentation in `lerna.json`

ntwb committed:
build: use `browserslist-config-wordpress` Browserslist shared config (#68)

* build: use `browserslist-config-wordpress` Browserslist shared config

* chore(babel-preset-default): switch to `@wordpress/browserslist-config`

* chore: fix merge conflict

* chore: regenerate `package-lock.json`

* Browserslit config: Add devDependency in the root repository

* Babel preset default: Add missing spaces around square brackets

gziolo committed:
Jest preset: Add default jest preset for WordPress development (#74)

* Jest preset: Add default jest preset for WordPress developmnent

* Jest preset: Use path to jest preset relative to root directory

gziolo committed:
Publish

 - @wordpress/babel-preset-default@1.0.3
 - @wordpress/jest-preset-default@1.0.0
 - @wordpress/scripts@1.0.2

gziolo committed:
Tests: Fix wrongly configured dependencies for Jest preset (#75)

gziolo committed:
Publish

 - @wordpress/jest-preset-default@1.0.1

gziolo committed:
Dependencies: Update Jest preset dependencies version (#76)

gziolo committed:
Build: Symlink all child packages instead of using versions published to npm (#77)

ntwb committed:
docs: add link to browserslist usage examples docs (#80)

* docs: update browserslist readme example usage

* docs (browserslist): Link to external configuration examples

* Update README.md

ntwb committed:
chore(browserslist): update Browserslist to v3.0.0

ntwb committed:
docs: ❤️  Code is Poetry

ntwb committed:
chore(browserslist): update Browserslist to v3.1.0

ntwb committed:
Publish

 - @wordpress/babel-preset-default@1.1.0
 - @wordpress/browserslist-config@2.1.0
 - @wordpress/jest-console@1.0.4
 - @wordpress/jest-preset-default@1.0.2

aduth committed:
Hooks: Avoid validating namespace in runHooks

Unnecessary because a hook cannot be registered with an invalid hook name, so it would not pass the subsequent condition to check that a hookset with corresponding name exists.

aduth committed:
Hooks: Use null prototype object with truthy access

https://jsperf.com/object-create-null-vs-hasownproperty

aduth committed:
Hooks: Simplify return first arg logic

Even if we don't intend to return value, no harm in assigning to args[ 0 ]

aduth committed:
Hooks: Assign hooks current initial value in creation

aduth committed:
Hooks: Avoid initializing hook history if not run

ntwb committed:
chore: update Lerna to 2.9.0

aduth committed:
Hooks: Add baseline benchmark

ntwb committed:
docs: use canonical _Code is Poetry_ image and center image using HTML

Props @rmccue.

gziolo committed:
Build: Use node script to symlink local npm dependencies (#79)

* Build: Use node script to symlink local npm dependencies

* Chore: Update symlink-or-copy to v1.2.0

* Chore: Added npm-run-all to simplify npm scripts

gziolo committed:
Scripts: Provide the default configuration for the `test` command (#83)

* Scripts: Provide the default configuration for the `test` command
It is used in the case when the project does not have a config for Jest or Babel

* Scripts: Add tests for utils to ensure they work properly

* Add back end-of-file new line

ntwb committed:
fix(scripts): rename script from `test-unit` to `test-unit-js` (#86)

gziolo committed:
Chore: Update Jest dependencies (#87)

gziolo committed:
Publish

 - @wordpress/a11y@1.0.6
 - @wordpress/autop@1.0.4
 - @wordpress/babel-preset-default@1.1.1
 - @wordpress/browserslist-config@2.1.1
 - @wordpress/dom-ready@1.0.3
 - @wordpress/hooks@1.1.4
 - @wordpress/jest-console@1.0.5
 - @wordpress/jest-preset-default@1.0.3
 - @wordpress/scripts@1.1.0
 - @wordpress/url@1.0.3

ntwb committed:
chore: remove `package-lock.json` files, lockfiles for apps, but not for packages

ntwb committed:
chore: add `package-lock.json` to `gitignore`

ntwb committed:
chore: add `package-lock=false` in new `.npmrc` files for each package

ntwb committed:
chore: add `"npmClientArgs": ["--no-package-lock"]` to Lerna `bootstrap` command

See https://github.com/lerna/lerna/issues/1235#issuecomment-365752506

ntwb committed:
Remove Lerna `npmClientArgs` option

aduth committed:
Merge pull request #85 from WordPress/update/run-hooks-perf

Hooks: Micro-optimize runHooks

aduth committed:
Publish

 - @wordpress/a11y@1.0.7
 - @wordpress/autop@1.0.5
 - @wordpress/babel-preset-default@1.1.2
 - @wordpress/browserslist-config@2.1.2
 - @wordpress/dom-ready@1.0.4
 - @wordpress/hooks@1.1.5
 - @wordpress/jest-console@1.0.6
 - @wordpress/jest-preset-default@1.0.4
 - @wordpress/scripts@1.1.1
 - @wordpress/url@1.0.4

ntwb committed:
Wrap filename in backticks

aduth committed:
Hooks: Fix undefined arguments on consecutive action callbacks

ntwb committed:
chore: add` jest-preset` keyword to `jest-preset-default`

aduth committed:
Package: Add `@wordpress/custom-templated-path-webpack-plugin` package

ntwb committed:
Add `publish:check` script to run `lerna updated`

aduth committed:
Merge pull request #90 from WordPress/fix/action-undef

Hooks: Fix undefined first argument on consecutive action callbacks

aduth committed:
Hooks: Correct CHANGELOG version to 1.1.6

aduth committed:
Publish

 - @wordpress/hooks@1.1.6
 - @wordpress/jest-preset-default@1.0.5
 - @wordpress/scripts@1.1.2

aduth committed:
Merge pull request #93 from WordPress/add/custom-templated-path-plugin

Package: Add `@wordpress/custom-templated-path-webpack-plugin` package

aduth committed:
Custom Templated Path Plugin: Remove debugging statement from tests

😬

ntwb committed:
docs: updated handbook URL

ntwb committed:
docs: update the Browserslist example repo URL (#98)

youknowriad committed:
WordPress i18n package: Initial commit (#96)

ntwb committed:
docs(i18n): Standardize `README.md` format

youknowriad committed:
Publish

 - @wordpress/babel-plugin-makepot@1.0.0
 - @wordpress/babel-preset-default@1.1.3
 - @wordpress/browserslist-config@2.1.3
 - @wordpress/custom-templated-path-webpack-plugin@1.0.1
 - @wordpress/i18n@1.0.0
 - @wordpress/scripts@1.1.3

adamsilverstein committed:
return a boolean from `hasHook` not the hook count

adamsilverstein committed:
simplify logic

adamsilverstein committed:
update tests to expect booleans from hasFilter/Action

adamsilverstein committed:
Clean up docs and return

adamsilverstein committed:
Merge pull request #102 from WordPress/fix/has-return-boolean

return a boolean from `hasHook` not the hook count

aduth committed:
babel-preset-default: Remove babel-plugin-lodash (#109)

aduth committed:
Publish

 - @wordpress/babel-preset-default@1.2.0
 - @wordpress/hooks@1.1.7
 - @wordpress/scripts@1.1.4

nerrad committed:
i18n: Support accumulatively registering additional locale data for domain (#105)

* allow for setting additional locale data for a domain accumulatively

* woops wrong object reference

* add test for accumulation of localeData

* repackage additional tests

* use more performant “in” check instead of lodash `_.has`

* use hasOwnProperty instead of in

* fix formatting for code style

* fix formatting

* use Object.assign instead of lodash.merge

* add new plural strings to tests

* simplify merging

* indent, indent

* i18n: Fix indentation

aduth committed:
i18n: Memoize dcnpgettext (#101)

* i18n: Memoize dcnpgettext

* i18n: Export dcnpgettext inline

aduth committed:
Publish

 - @wordpress/i18n@1.1.0

ryanwelcher committed:
Add WordCounter package (#10)

* Adding WordCounter package

* Simplifying test syntax

* export the count as default

* Creating a simpler API

* Refactor into a single function export

*  Modification based on review notes

* Complete refactor based on suggestions from @omarreiss

* Remove conditional check

* Moving each function into a new file. Exporting an object so we can use a single settings property and make the calls a little more sane

* Only importing the lodash method we need

* Using flow to chain the matchWords/matchCharacters inner function calls

* Adding readme

* Addressing some feedback on the PR

* Move to a simplier API

* Updates the README to match API changes

* Updates per review by @youknowriad

* Adds missing dockblock param

* Spacing issues

* Adds correct docblocks and fixes some whitespace issues

* Adding some whitespace as per review comment

pento committed:
Add prependHTTP() to @wordpress/url (#112)

pento committed:
Publish

 - @wordpress/url@1.1.0
 - @wordpress/wordcount@1.0.0

pento committed:
Add a publishConfig setting to wordcount's package.json, and fix the indenting. (#114)

pento committed:
Improve the publishing process documentation and error checking (#115)

aduth committed:
Add package: @wordpress/is-shallow-equal (#110)

* Add package: @wordpress/is-shallow-equal

* Ignore benchmark for codecov

No need to affect the codecoverage stats with a benchmarking script

* Define benchmark dependencies as devDependencies

Otherwise considered a proper required dependency from perspective of consuming project

aaronjorbin committed:
Add test for arrays being completely equal (#117)

* Add test for arrays being completely equal

It's a nearly useless test, but it does get is-shallow-equal to 100% test coverage
See https://codecov.io/gh/WordPress/packages/src/master/packages/is-shallow-equal/index.js#L48 for the one piece of code not currently tested.

* fix spacing

* add test for object copies

aduth committed:
is-shallow-equal: Fall back to strict equality for non-object-like (#116)

* Ensure inverse argument order has same test result

* is-shallow-equal: Fall back to strict equality for non-object-like

* is-shallow-equal: De-snark README.md

aduth committed:
is-shallow-equal: Add changelog entry for 1.0.1

aduth committed:
wordcount: Add changelog entry for 1.0.1

aduth committed:
Publish

 - @wordpress/is-shallow-equal@1.0.1
 - @wordpress/wordcount@1.0.1

adamsilverstein committed:
Improve hooks docs (#121)

adamsilverstein committed:
Add a missing `p` in the docs remove -> removep (#120)

ryanwelcher committed:
Fix/wordcount whitespace only error (#123)

* Addresses error thrown when the results of matchWord or matchCharacters returns null. Includes test for the case

* Changes as per @tofumatt

* Going full ternary

aduth committed:
is-shallow-equal: Use implicit index.js for main entry (#124)

aduth committed:
Add autop@1.0.5 CHANGELOG entry

aduth committed:
Add hooks@1.1.7 CHANGELOG entry

aduth committed:
Add is-shallow-equal@1.0.2 CHANGELOG entry

aduth committed:
Add wordcount@1.0.2 CHANGELOG entry

aduth committed:
Correct autop, hooks latest CHANGELOG versions

aduth committed:
Publish

 - @wordpress/autop@1.0.6
 - @wordpress/hooks@1.1.8
 - @wordpress/is-shallow-equal@1.0.2
 - @wordpress/wordcount@1.0.2

ntwb committed:
Add `@wordpress/npm-package-json-lint-config` package (#119)

* Add `@wordpress/npm-package-json-lint-config` package

* Standardize packages `package.json` format per `@wordpress/npm-package-json-lint`

* Add unused `npm-package-json-lint` rules for reference

* Update `npm-package-json-lint` to `3.0.0-alpha3`

* Add `valid-values-publishConfig` rule with `{"access": "public"}` option for _packages_ repo

* use tabs for indentation

* Update npm-package-json-lint to v3

* Update npmPackageJsonLintConfig config formatting

* Use latest `3.0.1` npm-package-json-lint release and remove `.npmpackagejsonlintrc.json`

* Use `lint:pkg-json` as the npmPkgJsonLint script task name

* fix `pretest` test script command

* Npm package.json lint config: address my own comments

* Add is-plain-object as devDependency

ntwb committed:
Add wordcount@1.0.3 CHANGELOG entry

ntwb committed:
Add scripts@1.1.5 CHANGELOG entry

ntwb committed:
Add npm-package-json-lint-config@1.0.0 CHANGELOG entry

ntwb committed:
Add jest-preset-default@1.0.5 CHANGELOG entry

ntwb committed:
Add jest-console@1.0.6 CHANGELOG entry

ntwb committed:
Add i18n@1.1.1 CHANGELOG entry

ntwb committed:
Add custom-templated-path-webpack-plugin@1.0.2 CHANGELOG entry

ntwb committed:
Add browserslist-config@2.1.3 CHANGELOG entry

ntwb committed:
Add babel-preset-default@1.2.1 CHANGELOG entry

ntwb committed:
Add babel-plugin-makepot@1.0.1 CHANGELOG entry

ntwb committed:
Add browserslist-config@2.1.4 CHANGELOG entry

ntwb committed:
Add jest-preset-default@1.0.6 CHANGELOG entry

ntwb committed:
Add jest-console@1.0.7 CHANGELOG entry

ntwb committed:
Add a custom Lerna publish commit message (#125)

ntwb committed:
chore(release): publish

 - @wordpress/babel-plugin-makepot@1.0.1
 - @wordpress/babel-preset-default@1.2.1
 - @wordpress/browserslist-config@2.1.4
 - @wordpress/custom-templated-path-webpack-plugin@1.0.2
 - @wordpress/i18n@1.1.1
 - @wordpress/jest-console@1.0.7
 - @wordpress/jest-preset-default@1.0.6
 - @wordpress/npm-package-json-lint-config@1.0.0
 - @wordpress/scripts@1.1.5
 - @wordpress/wordcount@1.0.3

gziolo committed:
Babel preset: Enable support for async generator functions (#126)

* Babel preset: Enabled support for async generator functions

* Babel preset: Add unit tests for async generator functions

* Babel preset: Update changelog

gziolo committed:
chore(release): publish

 - @wordpress/babel-preset-default@1.3.0
 - @wordpress/scripts@1.1.6

ntwb committed:
chore: update `codecov` to v3, resolves all `npm audit` issues. (#127)

gziolo committed:
Scripts: Add support for lint-pkg-json script (#128)

* Scripts: Add support for lint-pkg-json script

* Scripts: Update documentation for `lint-pkg-json` command

gziolo committed:
Scripts: Update changelog

gziolo committed:
chore(release): publish

 - @wordpress/scripts@1.2.0

pento committed:
Count when a hook that doesn't have any handlers is run. (#134)

Fixes #133.

pento committed:
chore(release): publish

 - @wordpress/hooks@1.2.0

gziolo committed:
Jest Console: Add new matchers for console.log and console.info (#137)

* Jest Console: Add new matchers for console.log and console.info

* Jest-console: Update CHANGELOG with braking changes details

omarreiss committed:
Add 'temp-packages/' from commit '39314f7068fb1ac5f7206ef01b0cf561b30bfe90'

git-subtree-dir: temp-packages
git-subtree-mainline: 04c0ea207efda81d98963c10e09c3c21705212fe
git-subtree-split: 39314f7068fb1ac5f7206ef01b0cf561b30bfe90

omarreiss committed:
Move packages from temporary directory to packages/

omarreiss committed:
Remove all the obsolete files from the old packages

omarreiss committed:
Synchronize package.json with the one from packages.

omarreiss committed:
Synchronize lerna.json with the one from packages

omarreiss committed:
Remove old packages build script

youknowriad commented:
I guess we should disable Allow "rebase and merge" for this PR ;)

ntwb commented:
> I guess we should allow "rebase and merge" for this PR ;)

Yes please, as per this weeks [#core-js Slack chat](https://wordpress.slack.com/archives/C5UNMSU4R/p1530018348000666) keeping the commit history I think is important

ntwb commented:
Notes:
- After merge, update URLs 
   - from `https://github.com/WordPress/packages` 
   - to `https://github.com/WordPress/gutenberg`

ntwb committed:
Add `is-shallow-equal/benchmark` to codecov ignore

ntwb commented on package.json:
```diff
@@ -161,11 +170,11 @@
 		"lint-pkg-json": "wp-scripts lint-pkg-json ./packages",
 		"package-plugin": "./bin/build-plugin-zip.sh",
-		"postinstall": "lerna bootstrap --hoist && npm run build:packages",
+		"postinstall": "lerna bootstrap --hoist && npm-run-all create-symlinks build:packages",
```
I think this needs to use `run-s` rather than `npm-run-all` to run _ sequential_ rather than _ parallel_ which the is the default cc @gziolo 

See https://www.npmjs.com/package/npm-run-all#cli-commands

gziolo replied:
It's sequential: https://github.com/mysticatea/npm-run-all/blob/HEAD/docs/npm-run-all.md#run-scripts-sequentially.
However, we should get rid of `npm-run-all`. We discussed it recently and agreed, that we already use another package for `parallel` execution and it's fine as is.

ntwb replied:
Thanks, removed in https://github.com/WordPress/gutenberg/pull/7556/commits/a268477c4d35f45b520e0e5c325989a7eff8b1b9

ntwb committed:
Remove npm-run-all

gziolo committed:
Packages: Synchronize dependencie to avoid having copies of the same libraries

gziolo committed:
Tests: Update test config to work with local packages

ntwb committed:
ESLint initial fixes via --fix

gziolo committed:
Build: Make Webpack build work again :)

gziolo committed:
Build: Fix the issue with babel makepot plugin

gziolo committed:
Packages: Fix all linter errors

gziolo commented:
> - Update package URLs 
>    - from `https://github.com/WordPress/packages` 
>    - to `https://github.com/WordPress/gutenberg`
> 
> 
> - Update dependancies to fix Lerna warnings
> 
> - Update `package-lock.json` file

Those items should be now resolved. In addition, I also had to fix `author` field in moved packages to align with the linter rules we have set.

gziolo committed:
Build: Use Lerna 3 for managing packages

It folows the setup of Lerna itself which demonstrates "the golden path of local file: specifiers at scale".
See: https://github.com/lerna/lerna/pull/1307.

gziolo commented:
There is still one blocker I discovered thanks to Travis. When you run `npm install` twice, those internally linked packages using `file:*` don't quite work. I'm not sure if the issue exists on Lerna's side, the setup we have or `package-lock.json` breaks it.

@ntwb or @ocean90 - any ideas?

gziolo commented:
This PR is now using Lerna 3 for managing packages with https://github.com/WordPress/gutenberg/pull/7556/commits/bc6aff37618856cc4d60df864574023953378bf6. It follows the setup of Lerna itself which demonstrates "the golden path of local file: specifiers at scale". See: lerna/lerna#1307.

It has a few advantages:
* We no longer need to sync versions of local dependencies, because we use relative paths and `file:` feature from `npm`.
* All dependencies are now hoisted by default and we no longer need to use `lerna bootstrap --hoist`.
* We just need to put a dependency in the `package.json` file in the root folder and it can be used in the code - see Webpack config as one of the examples.

gziolo commented on package.json:
```diff
@@ -96,17 +110,19 @@
 		"sass-loader": "6.0.7",
 		"sprintf-js": "1.1.1",
 		"style-loader": "0.20.3",
+		"symlink-or-copy": "1.2.0",
```
We probably don't need it since we use Lerna 3.x.

gziolo commented on package.json:
```diff
@@ -87,7 +97,11 @@
 		"espree": "3.5.4",
 		"extract-text-webpack-plugin": "4.0.0-beta.0",
+		"glob": "7.1.2",
+		"lerna": "3.0.0-beta.21",
```
@nerrad - I'm wondering if this can now safely be moved to devDependencies since it no longer enforces `postinstall` step as noted in #6718.

nerrad replied:
Ya with there being no `postinstall` it would be better to move it.

gziolo commented on test/unit/jest.config.json:
```diff
@@ -5,8 +5,9 @@
-		"@wordpress\\/(blocks|components|editor|utils|edit-post|viewport|core-data|core-blocks|nux)$": "$1",
-		"@wordpress\\/(api-request|blob|core-data|data|date|dom|deprecated|element|plugins|postcss-themes)$": "packages/$1/src"
+		"@wordpress\\/(blocks|components|editor|utils|edit-post|viewport|core-blocks|nux)$": "$1",
+		"@wordpress\\/(browserslist-config)$": "packages/$1",
```
We probably don't need this line anymore.

gziolo commented on test/unit/jest.config.json:
```diff
@@ -5,8 +5,9 @@
-		"@wordpress\\/(api-request|blob|core-data|data|date|dom|deprecated|element|plugins|postcss-themes)$": "packages/$1/src"
+		"@wordpress\\/(blocks|components|editor|utils|edit-post|viewport|core-blocks|nux)$": "$1",
+		"@wordpress\\/(browserslist-config)$": "packages/$1",
+		"@wordpress\\/(.*)$": "packages/$1/src"
```
@aduth @youknowriad - do we want to run our tests against `src`, `build` or `build-module`? What if we provide `src` locally and `build-module` on Travis?

youknowriad replied:
I think we should only run `src` for both.

gziolo replied:
We are all good then 👍

gziolo committed:
Merge remote-tracking branch 'origin/master' into add-packages

gziolo committed:
Packages: Fix configuration with newly introduced packages

gziolo committed:
Testing: Try to fix jest-console tests

gziolo committed:
Testing: Try to fix jest-console tests

gziolo commented:
Unit tests fail because of the following issue: https://github.com/facebook/jest/issues/4407:

> Using `.toThrowErrorMatchingSnapshot()` on code that prints error using ansi codes (i.e. using chalk) produces different snapshots if run locally or on CI.

There is a workaround proposed:
> https://github.com/joaogranado/jest-serializer-ansi

However, it still fails for us with a really interesting output:

> jest-console › console.error › toHaveErroredWith works when not called
> [1] 
> [1]     expect(value).toMatchSnapshot()
> [1]     
> [1]     Received value does not match stored snapshot 1.
> [1]     
> [1]     Compared values have no visual difference.
> [1] 
> [1]       47 | 					expect(
> [1]       48 | 						() => expect( console )[ matcherNameWith ]( message )
> [1]     > 49 | 					).toThrowErrorMatchingSnapshot();
> [1]       50 | 				} );
> [1]       51 | 
> [1]       52 | 				test( `${ matcherNameWith } works with many arguments that do not match`, () => {
> [1]       
> [1]       at Object.<anonymous> (packages/jest-console/src/test/index.test.js:49:8)

ntwb committed:
Merge branch 'add-packages' of github.com:WordPress/gutenberg into add-packages

ntwb committed:
jest-preset-default: Alphabetize dependencies

gziolo committed:
Testing: Use toThrowError matcher for jest-console

gziolo committed:
Merge remote-tracking branch 'origin/master' into add-packages

ntwb reviewed (approved):
:shipit:
